### PR TITLE
feat: configure use of prepared statements

### PR DESCRIPTION
**Pull request by yquansah**
Due to pgbouncer not supporting prepared statements when ran in transaction mode for pooling. We are going to introduce a configuration parameter to disable prepared statements if needed.

Completes FLI-432

### CODE REVIEW

File: internal/config/database.go [hunk 1]
```diff
@@ -27,16 +27,17 @@ const (
...
+	MaxIdleConn               int              `json:"maxIdleConn,omitempty" mapstructure:"max_idle_conn"`
+	MaxOpenConn               int              `json:"maxOpenConn,omitempty" mapstructure:"max_open_conn"`
+	ConnMaxLifetime           time.Duration    `json:"connMaxLifetime,omitempty" mapstructure:"conn_max_lifetime"`
+	Name                      string           `json:"name,omitempty" mapstructure:"name"`
+	User                      string           `json:"user,omitempty" mapstructure:"user"`
+	Password                  string           `json:"password,omitempty" mapstructure:"password"`
+	Host                      string           `json:"host,omitempty" mapstructure:"host"`
+	Port                      int              `json:"port,omitempty" mapstructure:"port"`
+	Protocol                  DatabaseProtocol `json:"protocol,omitempty" mapstructure:"protocol"`
+	PreparedStatementsEnabled bool             `json:"preparedStatementsEnabled,omitempty" mapstructure:"prepared_statements_enabled"`
```
Comment (markphelps) [inline internal/config/database.go:40]: we should update the schemas for cue and jsonschema for this config change.

- https://github.com/flipt-io/flipt/blob/main/config/flipt.schema.cue
- https://github.com/flipt-io/flipt/blob/main/config/flipt.schema.json

We probably are behind a bit tbh, especially with the experimental git/fs config changes. We should create some internal issues to update those (cc @GeorgeMac )

Comment (yquansah) [reply internal/config/database.go:40]: @markphelps Yeah agreed. I really want to get this thing reproducing locally. Not sure what could prevent it from not being reproduced locally.

Comment (GeorgeMac) [reply internal/config/database.go:40]: I reckon it might have something to do with numbers of replicas of things.
I can have a play this morning and see.

Comment (GeorgeMac) [reply internal/config/database.go:40]: I managed to replicate it by enabling it all with authentication enabled.
I got a hunch from the original issue:
```
postgres-flipt-1      | 2023-06-15T08:01:52Z	ERROR	unauthenticated	{"server": "grpc", "reason": "error retrieving authentication for client token", "error": "getting authentication by token: reading authentication: pq: prepared statement \"1\" already exists"}
postgres-flipt-1      | 2023-06-15T08:01:52Z	ERROR	unauthenticated	{"server": "grpc", "reason": "error retrieving authentication for client token", "error": "getting authentication by token: reading authentication: pq: prepared statement \"1\" already exists"}
```

For reference, this was my compose yaml:
```yaml
version: "3"

services:
  postgres:
    image: postgres:12
    networks:
      - flipt_network
    environment:
      - POSTGRES_DB=flipt
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=password

  pgbouncer:
    image: bitnami/pgbouncer
    networks:
      - flipt_network
    environment:
      - POSTGRESQL_USERNAME=postgres
      - POSTGRESQL_PASSWORD=password
      - POSTGRESQL_DATABASE=flipt
      - POSTGRESQL_HOST=postgres
      - PGBOUNCER_DATABASE=flipt
      - PGBOUNCER_POOL_MODE=transaction
      - PGBOUNCER_IGNORE_STARTUP_PARAMETERS=extra_float_digits

  flipt:
    build: .
    depends_on:
      - postgres
    ports:
      - "8080:8080"
    networks:
      - flipt_network
    volumes:
      - /Users/georgemac/login-with-google/config.yaml:/config.yml
    environment:
      - FLIPT_DB_URL=postgres://postgres:password@pgbouncer:6432/flipt?sslmode=disable
      - FLIPT_LOG_LEVEL=debug
      - FLIPT_META_TELEMETRY_ENABLED=false
    command: ["./tmp/wait-for-it.sh", "pgbouncer:6432", "--", "./flipt", "--force-migrate", "--config", "/config.yml"]

networks:
  flipt_network:
```

Then I also passed in a config yaml (you can see the absolute path in my machine) to a login with google config.
You could do the same with dex I reckon, I just had that lying around.

Comment (GeorgeMac) [reply internal/config/database.go:40]: @yquansah I pushed a change onto this branch. I got caught in the web of this one after I reproduced it. It wasn't as straightforward as I had hoped (surprise).

Checkout the latest commit. There is a long standing issue with lib/pq where it always does prepared statements unless you configure an option in conjunction with not using the statement cache.
@sashayakovtseva actually attempted the same fix in their reproduction (binary_parameters=yes), but I think we needed both statement cache disabled and this to make it complete.

**Review by GeorgeMac — APPROVED** [review]
Looks good as long as the tests pass 👍

**Review by markphelps — APPROVED** [review]
lgtm! thank you for the thorough reproduction

**Pull request merged**